src/CloudNative.CloudEvents.Amqp/AmqpClientExtensions.cs

```diff
@@ -113,7 +123,7 @@ public static Message ToAmqpMessage(this CloudEvent cloudEvent, ContentMode cont
                     properties = new Properties { ContentType = contentType.MediaType };
                     break;
                 case ContentMode.Binary:
-                    bodySection = SerializeData(cloudEvent.Data);
+                    bodySection = new Data { Binary = formatter.EncodeBinaryModeEventData(cloudEvent) };
```

Comment:
Should we validate that formatter is not null?

Reply:
This is something we need to do in all message converters, yes - along with calling `Validate` on the CloudEvents we receive and return. Will probably try to do that in a single sweep.